torch/optim/optimizer.py

```diff
@@ -15,6 +15,7 @@
 __all__ = ['Optimizer', 'register_optimizer_step_pre_hook', 'register_optimizer_step_post_hook']
 _global_optimizer_pre_hooks: Dict[int, Callable] = OrderedDict()
 _global_optimizer_post_hooks: Dict[int, Callable] = OrderedDict()
+_foreach_supported_types = [torch.Tensor, torch.nn.parameter.Parameter]
```

Comment:
nit: make it a tuple instead so it's not mutated accidentally mutated?

Reply:
This is actually intentional! This PR was a realization that the defaulting was way too conservative and did not allow models to be included for faster foreach. We want to allow users to be able to add their own subclasses here if they know that foreach will not break on those tensor subclasses.

Reply:
@janeyx99 What is the recommended way to add to this list? Should we directly append like the following in trainer code:
```
torch.optim.optimizer._foreach_supported_types.append(...)
```

Reply:
@janeyx99 Have the same qq @awgu raised. Do you know how users are adding their own subclasses to the list? 
Due to circular dependency, instead of adding DTensor here, I am just trying out what Andrew suggested first. Seems appending tensor subclass to the list has no effect at later stage so I am wondering how users are doing this if you are aware.

Reply:
You may be the first user to try this--what Andrew suggests seems okay for now barring the fact that it's a private API. Once we want to productionize this, we should work together to make this an exposed public API.

Reply:
It sounds like appending is not working?

Reply:
> It sounds like appending is not working?

My bad! False information. Just double check append seems working fine on the unit test locally!